torch/utils/cpp_extension.py

```diff
@@ -917,6 +919,42 @@ def _prepare_ldflags(extra_ldflags, with_cuda, verbose):
     return extra_ldflags
 
 
+def _get_cuda_arch_flags(cflags=None):
+    # If cflags is given, there may already be user-provided arch flags in it
+    # (from `extra_compile_args`)
+    if cflags is not None:
+        for flag in cflags:
+            if 'arch' in flag:
+                return []
+
+    # The default is sm_30 for CUDA 9.x and 10.x
+    # First check for an env var (same as used by the main setup.py)
+    # Can be one or more architectures, e.g. '6.1' or "3.5 5.2 6.0 6.1 7.0+PTX"
+    # See cmake/Modules_CUDA_fix/upstream/FindCUDA/select_compute_arch.cmake
+    arch_list = os.environ.get('TORCH_CUDA_ARCH_LIST', None)
+
+    all_arches = {'3.5': '35', '5.2': '52', '6.0': '60', '6.1': '61',
+                  '7.0': '70', '7.0+PTX': '70',  '7.2': '72', '7.2+PTX': '72',
+                  '7.5': '75', '7.5+PTX': '75',
+                  'Kepler': '52', 'Tegra': '52', 'Maxwell': '52',
+                  'Pascal': '60', 'Volta': '70', 'Turing': '75'}
```

Comment:
I didn't check the correctness of this table.

Reply:
for reference, here is the CMake code I based it on:
```
  list(APPEND CUDA_KNOWN_GPU_ARCHITECTURES "Kepler+Tegra" "Kepler+Tesla" "Maxwell+Tegra")
  list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "5.2")

  list(APPEND CUDA_KNOWN_GPU_ARCHITECTURES "Pascal")
  list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "6.0" "6.1")

  list(APPEND CUDA_KNOWN_GPU_ARCHITECTURES "Volta")
  list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "7.0" "7.0+PTX")
  list(APPEND CUDA_ALL_GPU_ARCHITECTURES "7.0" "7.0+PTX" "7.2" "7.2+PTX")

  list(APPEND CUDA_KNOWN_GPU_ARCHITECTURES "Turing")
  list(APPEND CUDA_COMMON_GPU_ARCHITECTURES "7.5" "7.5+PTX")
  list(APPEND CUDA_ALL_GPU_ARCHITECTURES "7.5" "7.5+PTX")
```

Sometimes I had to choose, e.g. `Pascal` can be `6.0` or `6.1`, so I chose the lowest ones.